rewrites for tt-explorer content, architecture page needs more work (#5504)
tenstorrent/tt-mlir: docs/src/tt-explorer/usage-api.md

## TT-Adapter

This section provides a reference for the TT-Adapter REST API, as well as an explanation of how TT-Explorer creates an extensible API on top of Google's Model Explorer.

### Building an API using Model Explorer

Google's Model Explorer defines the endpoint `/apipost/v1/send_command` as a generic command-receiving mechanism. This endpoint lets the server accept requests to perform operations on a model. Specifically, it:
* Accepts JSON commands describing an action (for example, convert a model or run an analysis).
* Parses commands to ensure they follow the correct format.
* Forwards the command to the installed adapter—here, TT-Adapter.
* Wraps the output in a standard JSON response that the UI (TT-Explorer’s interface) can interpret.

This endpoint provides the main communication channel between the client and server. Commands sent here are executed by the specified adapter and return a structured “adapter response,” allowing developers to interact programmatically with models through TT-Explorer.

#### Flow Overview

> This section provides a general overview of how commands flow between the client and server on your local machine.

Reviewer: <!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent. -->
**suggestion:** There is no difference in architecture for a local server vs a hosted server, so We can remove the "on your local machine" part.

```suggestion
This section provides a general overview of how commands flow between the client and server.
```

Author: Updated

![Workflow Diagram for TT-Explorer](../images/tt-explorer/explorer-api-flow.png)

1. UI (Client) sends commands via HTTP to the Local Server.

2. Server (Backend) forwards commands to the TT-Adapter for execution.

3. TT-Adapter executes the model / graph operations locally (may reuse upstream code).

4. Adapter produces a raw response → Server processes it via `convert_adapter_response`.

5. Server returns JSON-wrapped results to the UI, which renders graphs, overlays, and status.

> Everything happens on your local machine.

Reviewer: <!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent. -->
**suggestion:** Same here, I think we can remove this as it may be misleading. The execution happens in the server, it is only local if the server and client are on the same machine.

```suggestion
```

Author: Added

    pass
```

> Information does not persist between requests—each request gets a new instance of TT-Adapter. As all requests to the server are stateless, each command-handling function receives input only through its parameters (`model_path` and `settings`). Any data needed for execution must be passed in the request, and the client is responsible for keeping track of important information such as the path of an uploaded model or the locations of artifacts produced by the server.

Reviewer: **issue:** The TT-Adapter instance is the same, but the nature of HTTP requests is that they are stateless.

**nitpick**: Formatting.

```suggestion
Information does not persist between requests due to the stateless nature of HTTP requests. Therefore each command-handling function receives input only through its parameters (`model_path` and `settings`).

Any data needed for execution must be passed in the request, and the client is responsible for keeping track of important information such as the path of an uploaded model or the locations of artifacts produced by the server.
```

Author: Not sure what's being seen. Looks okay in the preview here.

- Is a [double precision floating point number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number#number_encoding)
- Is not [`NaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN#observably_distinct_nan_values)

> - Is greater than or equal to `0`

Reviewer: Nice catch!
